clicommand/bootstrap.go

```diff
@@ -376,6 +387,34 @@ var BootstrapCommand = cli.Command{
 	},
 	Action: func(c *cli.Context) error {
 		ctx := context.Background()
+
+		// Surprise! Before doing anything else, even loading the other
+		// flags/config, we connect the socket to get env vars. These are
+		// normally present in the environment if the bootstrap was forked,
+		// directly, but because in k8s land we use containers to separate the
+		// agent and its executors, such vars won't be available unless we do
+		// this.
+		var k8sClient *kubernetes.Client
+		if c.Bool(KubernetesExecFlag.Name) {
+			k8sClient = &kubernetes.Client{ID: c.Int("kubernetes-container-id")}
+
+			rtr := roko.NewRetrier(
+				roko.WithMaxAttempts(7),
+				roko.WithStrategy(roko.Exponential(2*time.Second, 0)),
+			)
+			regResp, err := roko.DoFunc(ctx, rtr, func(rtr *roko.Retrier) (*kubernetes.RegisterResponse, error) {
+				return k8sClient.Connect(ctx)
+			})
+			if err != nil {
+				return fmt.Errorf("error connecting to kubernetes runner: %w", err)
+			}
+			for n, v := range env.FromSlice(regResp.Env).Dump() {
+				if err := os.Setenv(n, v); err != nil {
+					return err
+				}
+			}
```

Comment:
The only issues I can think of for this are: 

1. now many environment variables set by our k8s controller as part of the pod spec will be entirely useless. We might want to eliminate those logic in k8s stack.
2. Is there any environment variables that agent container have access but bootstrap containers don't/shouldn't?

Reply:
1. Probably!
2. In theory, everything the agent container passes this way to the bootstrap container would, if not using the k8s stack, be set specifically on the bootstrap subprocess forked by the agent. So I don't imagine there's anything the bootstrap shouldn't have access to. Definitely something to think about.

Reply:
Point 2 was a good callout - after thinking on it for a bit, some of the vars normally set by the agent would interfere with the variables set by the k8s stack if they were set after the bootstrap is forked.